[IMP] website: complete google search console doc (#9767)
odoo/documentation: content/applications/websites/website/configuration/google_search_console.rst

valuable insights into how Google views and interacts with your site, helping you optimize its
performance.

> There are five ways to verify your site ownership:

Author: Hi dear @auva-odoo, just updated it!

pane. Once again, if the verification does not work immediately, wait a few hours or a day and try
again, as the change may take some time to take effect. Just choose the unverified property from the
:guilabel:`Properties` drop-down and it will try to verify you automatically.

.. important:: To stay verified, do not remove the DNS Record even after verification succeeds.

.. _GSC-HTML-File-Upload:

HTML File Upload
----------------

This method involves uploading an HTML file provided by Google to the root directory of your
website. Google verifies ownership by checking for this file. You will typically download the HTML
verification file from Google Search Console, and then upload it to your website. Here, you just
have to copy-paste the code in the appropriate field in the Website Settings. You will need to have
permission to upload the file to your website where it can be accessed by a browser. Once you added
your website URL under the URL prefix option and clicked on :guilabel:`continue`, expand the HTML
file section where you will find a file :guilabel:`Download` button.

.. image:: google_search_console/html-file-download.png
   :alt: HTML file download

> Download your HTML verification file and upload it to the root directory of the website you want to
> verify.

Reviewer: ```suggestion
#. Download your HTML verification file and upload it to the root directory of the website you want to
verify.
#. Open the file and copy the verification code, e.g., `google123abc.html`.
```
- Do people generally know how to access the root directory of their website? (sorry, I am *not* an expert on this at all)
- I assume when you say on line 99 "paste the code", you mean the one we can find when we open the html file? I added it here otherwise people might get confused where they need to copy the code from. Feel free to change this if I'm wrong :slightly_smiling_face:

.. image:: google_search_console/open-copy-html-file.png

>    :alt: Open and copy html file

Reviewer: I would crop this screenshot so we don't see the "other verification methods", it would look cleaner and is not needed IMO

> Then, access your Odoo database, go to :menuselection:`Website --> Configuration --> Settings`,
> and enable :guilabel:`Google Search Console` in the :guilabel:`SEO` section. Paste the
> google123abc.html code in the dedicated field.

Reviewer: ```suggestion
#. In your Odoo database, go to :menuselection:`Website --> Configuration --> Settings` and enable :guilabel:`Google Search Console` in the :guilabel:`SEO` section. Paste the
verification code (e.g., `google123abc.html`) in the dedicated field.
```

.. image:: google_search_console/paste-html-code-settings.png
   :alt: Paste html code in Odoo

> Once you complete this step, click :guilabel:`Verify` in Search Console. If you perform the steps
> above correctly, verification should work immediately.

Reviewer: ```suggestion
#. In Google Search Console, click :guilabel:`Verify`. If you performed the steps above correctly, verification should be done immediately.
```

Five ways to verify your site ownership
=======================================

.. _GSC-DNS-Record:

DNS Record
----------

With this method, you add a DNS TXT record to your domain's DNS settings. Google provides you with a
unique string to add as a TXT record. Once added, Google verifies ownership by checking the DNS
settings.
Start by copying the text record from the dialog google-site-verification=123aBc… after entering
your domain.
After you change your DNS record, go back to Search Console and click Verify in the verification pane.
Once again, if the verification does not work immediately, wait a few hours or a day and try again,
as the change may take some time to take effect. Just choose the unverified property from the
:guilabel:`Properties` drop-down and it will try to verify you automatically.

.. important:: To stay verified, do not remove the DNS Record even after verification succeeds.

.. _GSC-HTML-File-Upload:

HTML File Upload
----------------

This method involves uploading an HTML file provided by Google to the root directory of your website.
Google verifies ownership by checking for this file. You will typically download the HTML verification

> file from Google Search Console, and then upload it to your website using FTP or file manager provided

Reviewer 2: Hi @masp-odoo , Actually. for this part, could you remove it?
" using FTP or file manager provided
by your web hosting service. Odoo is the website builder that allows web hosting service on
:doc:`/administration/odoo_online` and :doc:`Odoo.sh </administration/odoo_sh/overview/introduction>`.
If you are using Odoo Online, it is a :abbr:`SaaS (Software as a service)` solution, meaning that it
is not :abbr:`FTP (File Transfert Protocol)`. Odoo Online provides private databases which are fully
managed and hosted by Odoo"

The parts before and after are fine. This one looks unnecessary because there is no option in Odoo called "FTP". In addition, the file upload through copy/paste of the google123abc.html code is actually type http.
https://github.com/odoo/odoo/blob/dcd426df41db9200b95d04ba664441053281db78/addons/website/controllers/main.py#L784
It's not relevant to describe FTP here for Odoo. Let's also not talk about Odoo Online and Odoo SH to maintain the topic about the HTML File Upload.

Thank you.

managed and hosted by Odoo. Here, you just have to copy-paste the code in the appropriate field in
the Website Settings. You will need to have permission to upload the file to your website where it
can be accessed by a browser. Once you added your website URL under the URL prefix option and clicked
on :guilabel:`continue`, expand the HTML file section where you will find a file :guilabel:`Download`
button.

.. image:: google_search_console/html-file-download.png
   :alt: HTML file download

Download your HTML verification file and upload it to the root directory of the website you want to
verify.

.. image:: google_search_console/open-copy-html-file.png
   :alt: Open and copy html file

Then, access your Odoo database, go to :menuselection:`Website --> Configuration --> Settings`,
and enable :guilabel:`Google Search Console` in the :guilabel:`SEO` section. Paste the google123abc.html
code in the dedicated field.

.. image:: google_search_console/paste-html-code-settings.png
   :alt: Paste html code in Odoo

Once you complete this step, click :guilabel:`Verify` in Search Console. If you perform the steps
above correctly, verification should work immediately.

> .. example:: If you want to verify www.example.com, then the root directory is the homepage. If you

Reviewer 2: For this example, I recommend you to remove. After reading it, it is unnecessary to explain about /blog/ path because the Google Search Console option in Odoo verifies at the path /google<string(length=16):key>.html
It's not at /blog/google<string(length=16):key>.html
https://github.com/odoo/odoo/blob/dcd426df41db9200b95d04ba664441053281db78/addons/website/controllers/main.py#L784
Customers look to verify *domain* using site verification not the *domain's path* using site verification.

Author: Done! Thank you for the explanation :)

   Before verifying, ensure that your homepage has the Google Analytics tracking code in the head
   section of the page. Although your Analytics tracking code might work for Analytics in the body
   section as well, for Search Console verification it must be in the head section.

.. seealso::
   :doc:`../reporting/analytics`

Open your homepage in a browser, look at the page source code and confirm that the tag is present in
the head tag of the page.

After you click :guilabel:`Add property`, add your website URL under the URL prefix option and click
:guilabel:`Continue`.
If you perform the steps above correctly, verification should work immediately.

.. note:: This works for the verification process only. No data will be collected here, configuring
   Google Analytics is an entirely separate process.

.. _GSC-Google-Tag-Manager-Container-Snippet:

Google Tag Manager Container Snippet
------------------------------------

If you use Google Tag Manager to manage tags on your website, you can verify ownership of your site
by adding a specific HTML tag through the Google Tag Manager interface.

.. _GSC-Google-Sites:

> Google Sites

Reviewer 2: For Google SItes, I recommend removing this section as well because Google Sites is Odoo Website's competitor.
https://sites.google.com/

Author: Sorry that I missed that too, thanks for telling I removed it!

.. _GSC-DNS-Record:

DNS Record
----------

With this method, you add a DNS TXT record to your domain's DNS settings. Google provides you with a
unique string to add as a TXT record. Once added, Google verifies ownership by checking the DNS
settings.
Start by copying the text record from the dialog google-site-verification=123aBc… after entering
your domain.
After you change your DNS record, go back to Search Console and click Verify in the verification pane.
Once again, if the verification does not work immediately, wait a few hours or a day and try again,
as the change may take some time to take effect. Just choose the unverified property from the
:guilabel:`Properties` drop-down and it will try to verify you automatically.

.. important:: To stay verified, do not remove the DNS Record even after verification succeeds.

.. _GSC-HTML-File-Upload:

HTML File Upload
----------------

This method involves uploading an HTML file provided by Google to the root directory of your website.
Google verifies ownership by checking for this file. You'll typically download the HTML verification
file from Google Search Console, and then upload it to your website using FTP or file manager

> provided by your web hosting service. You will need to have permission to upload the file to your

Reviewer 2: In This sentence "You'll typically....provided by your web hosting service". I recommend rewording it or adding to this line because Odoo is the website builder that allows web hosting service on SaaS and SH. Clients who struggle with Google Site Verification are not that Tech Savvy. They are more curious about wanting to know how to verify their website on Odoo easily.
I suggest adding or rewording to highlight Odoo. This is also good to prevent tech blind clients from sending in tickets saying "how do I upload to website using FTP?" when they are on SaaS...
